lib/Backend/BackwardPass.cpp

```diff
@@ -2735,6 +2846,168 @@ BackwardPass::ProcessBlock(BasicBlock * block)
 #endif
 }
 
+bool 
+BackwardPass::CanDeadStoreInstrForScopeObjRemoval(Sym *sym) const
+{
+    if (tag == Js::DeadStorePhase && this->currentInstr->IsStackArgsEnabled())
+    {
+        Func * currFunc = this->currentInstr->m_func;
+        switch (this->currentInstr->m_opcode)
+        {
+            case Js::OpCode::LdHeapArguments:
+            case Js::OpCode::LdHeapArgsCached:
+            case Js::OpCode::LdLetHeapArguments:
+            case Js::OpCode::LdLetHeapArgsCached:
+            {
+                if (this->currentInstr->GetSrc1()->IsScopeObjOpnd(currFunc))
+                {
+                    return true;
+                }
+                break;
+            }
+            case Js::OpCode::LdSlot:
+            {
+                if (sym && IsFormalParamSym(currFunc, sym))
+                {
+                    return true;
+                }
+                break;
+            }
+            case Js::OpCode::CommitScope:
+            {
+                return this->currentInstr->GetSrc1()->IsScopeObjOpnd(currFunc);
+            }
+            case Js::OpCode::BrFncCachedScopeEq:
+            case Js::OpCode::BrFncCachedScopeNeq:
+            {
+                return this->currentInstr->GetSrc2()->IsScopeObjOpnd(currFunc);
+            }
+        }
+    }
+    return false;
+}
+
+/*
+* This is for Eliminating Scope Object Creation during Heap arguments optimization.
+*/
+bool
+BackwardPass::DeadStoreOrChangeInstrForScopeObjRemoval()
+{
+    IR::Instr * instr = this->currentInstr;
+    Func * currFunc = instr->m_func;
+
+    if (this->tag == Js::DeadStorePhase && instr->IsStackArgsEnabled() && !IsPrePass())
+    {
+        switch (instr->m_opcode)
+        {
+            /*
+            *   This LdSlot loads the formal from the formals array. We replace this a Ld_A <ArgInSym>.
+            *   ArgInSym is inserted at the beginning of the function during the start of the deadstore pass- for the inliner.
```

Comment:
I meant in the comment, change 'inliner' to 'top func' since an inliner may not be top func

---
In reply to: [67800979](https://github.com/Microsoft/ChakraCore/pull/1135#discussion_r67800979) [](ancestors = 67800979,67741254)